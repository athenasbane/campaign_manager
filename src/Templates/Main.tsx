import { Container, Grid } from "@mui/material";
import Navbar from "Components/Molecule/Navbar/Navbar";
import Draw from "Components/Molecule/Draw/Draw";
import { ReactNode } from "react";

export interface IMainTemplateProps {
  children: ReactNode;
  NavbarProps: {
    onMenuButtonClick: () => void;
  };
  DrawProps: {
    open: boolean;
    closeModal: () => void;
    openSingleModal: () => void;
  };
}

export default function MainTemplate({
  NavbarProps,
  DrawProps,
  children,
}: IMainTemplateProps) {
  return (
    <>
      <Container maxWidth="lg">
        <Grid container direction={"column"}>
          <Grid item sx={{ position: "fixed", zIndex: 100, left: 0 }}>
            <Navbar onMenuButtonClick={NavbarProps.onMenuButtonClick} />
          </Grid>
          <Grid item sx={{ mt: 30 }}>
            {children}
          </Grid>
        </Grid>
      </Container>
      <div>
        <Draw
          open={DrawProps.open}
          closeModal={DrawProps.closeModal}
          openSingleModal={DrawProps.openSingleModal}
        />
      </div>
    </>
  );
}
