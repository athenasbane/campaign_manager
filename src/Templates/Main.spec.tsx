import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Main from "../Templates/Main";

describe("Template - Main", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <Main
        NavbarProps={{ onMenuButtonClick: jest.fn() }}
        DrawProps={{
          open: false,
          closeModal: jest.fn(),
          openSingleModal: jest.fn(),
        }}
      >
        <div>Test</div>
      </Main>,
      { wrapper: BrowserRouter }
    );

    expect(container).toMatchSnapshot();
  });
});
