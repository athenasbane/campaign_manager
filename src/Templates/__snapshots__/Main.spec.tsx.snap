// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template - Main should match snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-zs5l8h-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-1i9rmbe"
        >
          <div
            class="MuiGrid-root MuiGrid-container css-hwacae-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
            >
              <a
                href="/"
                style="text-decoration: none; color: white;"
              >
                <h3
                  class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
                >
                  Tordenhelm
                </h3>
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
                data-testid="menu__button"
                tabindex="0"
                type="button"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  Menu
                </p>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-1kyi2hg-MuiGrid-root"
      >
        <div>
          Test
        </div>
      </div>
    </div>
  </div>
  <div>
    <div>
      <div
        class="PrivateSwipeArea-root css-17j2vre"
        style="height: 20px;"
      />
    </div>
  </div>
</div>
`;
